llvm/lib/Target/AMDGPU/AMDGPUAlwaysInlinePass.cpp

```diff
@@ -166,6 +168,6 @@ ModulePass *llvm::createAMDGPUAlwaysInlinePass(bool GlobalOpt) {
 
 PreservedAnalyses AMDGPUAlwaysInlinePass::run(Module &M,
                                               ModuleAnalysisManager &AM) {
-  alwaysInlineImpl(M, GlobalOpt);
-  return PreservedAnalyses::all();
+  const bool Changed = alwaysInlineImpl(M, GlobalOpt);
+  return Changed ? PreservedAnalyses::none() : PreservedAnalyses::all();
```

Comment:
This could really preserves most but we should be deleting this pass eventually

Reply:
Good to note this. We can be conservative for now even though I think this may preserve all or most analyses. Perhaps to define a `most` set would be helpful.

Always Inline pass only replaces aliases with functions, and then adds attributes to the function, so the pass could use that set if it stays around. Regardless, there are other passes that could benefit from it as well.

Will submit as it is now. Thank you.